### PR TITLE
Gapless aware seek handling

### PR DESCRIPTION
**Pull request by adamcik**
Fixes:  #312, #1304 (partially), #1305

Changelog to follow once I've reviewed the diff again.

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -16,6 +16,11 @@ Core API
 - Start ``tlid`` counting at 1 instead of 0 to keep in sync with MPD's
   ``songid``.
 
+- `get_time_position` now returns the seek target while a seek is in progress.
```
Comment (jodal) [inline docs/changelog.rst]: Either you need double backtick quotes or use `:meth:~mopidy.core.PlaybackController.get_time_position`.

File: docs/changelog.rst [hunk 2]
```diff
@@ -16,6 +16,11 @@ Core API
...
+  This gives better results than just failing the position query.
+
+  (Fixes: :issue:`312` PR: :pr:`1346`)
```
Comment (jodal) [inline docs/changelog.rst]: Put this in the same paragraph as the rest.

We don't have support for `:pr:` yet, it was only suggested on IRC. Feel free to add it to `docs/conf.py`.

File: docs/changelog.rst [hunk 3]
```diff
@@ -83,6 +88,11 @@ Gapless
 - Tests have been updated to always use a core actor so async state changes
   don't trip us up.
 
+- Seek events are now triggered when the seek completes. Further changes have
```
Comment (jodal) [inline docs/changelog.rst]: Sounds like we didn't trigger that event at all before. Need to explain what the old behavior was.

File: docs/changelog.rst [hunk 4]
```diff
@@ -83,6 +88,11 @@ Gapless
...
+  been made to make seek work correctly for gapless related corner cases.
+
+  (Fixes: :issue:`1305` PR: :pr:`1346`)
```
Comment (jodal) [inline docs/changelog.rst]: Same comments as above.

**Comment by jodal** [comment]
Reviewed commit by commit. LGTM.

**Comment by adamcik** [comment]
Done.

File: docs/changelog.rst [hunk 5]
```diff
@@ -83,6 +87,11 @@ Gapless
 - Tests have been updated to always use a core actor so async state changes
   don't trip us up.
 
+- Seek events are now triggered when the seek completes. Previously the event
+  was emitted when the seek was requested, not when it completed. Further
+  changes have been made to make seek work correctly for gapless related corner
+  cases. (Fixes: :issue:`1305` PR: :pr:`1346`)
```
Comment (jodal) [inline docs/changelog.rst]: "Same comments as above" means that `:pr:` must be replaced with `:issue:` here as well. ;-)

**Comment by adamcik** [comment]
Done

**Pull request merged**